src/app/clusters/webrtc-requestor-server/webrtc-requestor-server.h

```diff
@@ -0,0 +1,169 @@
+{
+public:
+    WebRTCRequestorDelegate() = default;
+
+    virtual ~WebRTCRequestorDelegate() = default;
+
+    /**
+     * @brief
+     *   Handles the Offer command received by the server.
+     *
+     * @param[in]  sessionId Current session ID.
+     * @param[in]  sdpOffer  SDP offer received.
+     * @param[in]  iceServers ICE Servers received to generated public ICE candidates.
+     * @param[in]  iceTransportPolicy Allowed ICE candidate policy (e.g.; 'all', 'relay') to generate ICE candidates
+     * @param[in]  peerNodeId The Node ID of the provider.
+     * @param[in]  peerFabricIndex The FabricIndex of the provider.
+     *
+     * @param[out] outSession New session struct is created with the new session details received.
+     *
+     * @return CHIP_ERROR
+     *   - Returns error if the session is invalid or the candidates cannot be processed
+     */
+    virtual CHIP_ERROR HandleOffer(uint16_t sessionId, const chip::CharSpan & sdpOffer,
+                                   const Optional<DataModel::DecodableList<ICEServerStruct>> & iceServers,
+                                   const Optional<chip::CharSpan> & iceTransportPolicy, NodeId peerNodeId,
+                                   FabricIndex peerFabricIndex, WebRTCSessionStruct & outSession) = 0;
+    /**
+     * @brief
+     *   Handles the Answer command received by the server.
+     *
+     * @param[in]  sessionId Current session ID.
+     * @param[in]  sdpAnswer  SDP answer received.
+     *
+     * @return CHIP_ERROR
+     *   - Returns error if the session is invalid or the candidates cannot be processed
+     */
+    virtual CHIP_ERROR HandleAnswer(uint16_t sessionId, const chip::CharSpan & sdpAnswer) = 0;
+    /**
+     * @brief
+     *   Called when the server receives the ICECandidates command.
+     *
+     * @param[in] sessionId  Current session ID.
+     * @param[in] candidates List of ICE candidate strings.
+     *
+     * @return CHIP_ERROR
+     *   - Returns error if the session is invalid or the candidates cannot be processed
+     */
+    virtual CHIP_ERROR HandleICECandidates(uint16_t sessionId, const std::vector<std::string> & candidates) = 0;
+    /**
+     * @brief
+     *   Called when the server receives the End command.
+     *
+     * @param[in] sessionId  Current session ID to end.
+     * @param[in] reasonCode Reason to end the session.
+     *
+     * @return CHIP_ERROR
+     *   - Returns error if the session is invalid or the candidates cannot be processed
+     */
+    virtual CHIP_ERROR HandleEnd(uint16_t sessionId, WebRTCEndReasonEnum reasonCode) = 0;
+
+protected:
+    friend class WebRTCRequestorServer;
+
+    WebRTCRequestorServer * mWebRTCRequestorServer = nullptr;
+
+    // sets the Chime Server pointer
+    void SetWebRTCRequestorServer(WebRTCRequestorServer * WebRTCRequestorServer) { mWebRTCRequestorServer = WebRTCRequestorServer; }
+    WebRTCRequestorServer * GetWebRTCRequestorServer() const { return mWebRTCRequestorServer; }
```

Comment:
Those two APIs are not needed. The cluster server is initialed from the app, not from its delegate. See other clusters's implementations.